Improvements for password and Mastodon server input.

Spaxe opened the pull request:
This PR fixes #21. In addition, password should be input via `getpass.getpass()`. It is very terrible when you're typing password in plaintext, stored in terminal. This PR fixes that.

Spaxe committed:
Improvements for password and Mastodon server input.

Spaxe commented:
This PR also adds the choice of Mastodon server other than `mastodon.social` as part of the command prompt.

Spaxe committed:
Added storage of Mastodon server base URL

tobico commented on MastodonToTwitter.py:
```diff
@@ -39,7 +38,7 @@
 
 # Media regex, to trim media URLs
-MEDIA_REGEXP = re.compile(re.escape(MASTODON_BASE_URL.rstrip("/")) + "\/media\/(\w)+(\s|$)+")
+MASTODON_BASE_URL = ''
```
Is this declaration needed? Also, it looks like the comment is out of date

Spaxe replied:
I wasn't sure at first, so I left it in. Checking it now, it looks like it's declared separately on creating a new app, or loading from secrets. Removed.

tobico commented on MastodonToTwitter.py:
```diff
@@ -114,8 +113,28 @@ def calc_expected_status_length(status, short_url_length = 23):
+
+        print("\n")
+        if os.path.isfile("mtt_mastodon_server.secret"):
+            print("You already have mastodon server set up, so we're skipping that step.")
```
"Mastodon" should start with a capital letter.

tobico commented on MastodonToTwitter.py:
```diff
@@ -114,8 +113,28 @@ def calc_expected_status_length(status, short_url_length = 23):
+        if os.path.isfile("mtt_mastodon_server.secret"):
+            print("You already have mastodon server set up, so we're skipping that step.")
+        else:
+            print("Recording mastodon server...")
```
Same here

Spaxe committed:
Capitalisation for Mastodon

Spaxe committed:
Removed excessive variable declaration

Spaxe commented:
Thanks @tobico, addressed your comments.

halcy commented on MastodonToTwitter.py:
```diff
@@ -114,8 +113,15 @@ def calc_expected_status_length(status, short_url_length = 23):
+        if MASTODON_BASE_URL == '':
+            # The Mastodon instance base URL. By default, https://mastodon.social/
+            MASTODON_BASE_URL = "https://mastodon.social"
+        elif not MASTODON_BASE_URL.startswith('https://'):
```
If somebody enters something starting with http:// this breaks, I think.

halcy commented on MastodonToTwitter.py:
```diff
@@ -114,8 +113,15 @@ def calc_expected_status_length(status, short_url_length = 23):
 
     mastodon_works = False
     while mastodon_works == False:
+        MASTODON_BASE_URL = input("Mastodon server (e.g. mastodon.social): ").strip()
```
Should probably indicate that just pressing enter results in default.

halcy reviewed (changes requested):
There's two things I'd like to be changed before I merge (or I can change them after), see comments

Spaxe committed:
Clearer input direction to remove https:// in input, and enter for defaults

Spaxe commented:
Thanks @halcy. I added `https://` in front of the prompt, so user will know not to type the protocol (and if they did, the connection will fail and make them redo it). Added "press Enter for" default also.

Spaxe committed:
Adding the actual protocol into input

Spaxe reviewed (commented):
Test comment.

Spaxe commented:
Doesn't look like I can mark this as "Requested Changes Made", so it's up to you now. :)

halcy commented:
I can contribute to open source projects while waiting in line at the post office. Modern technology is incredible.

Thanks for your work!

Spaxe commented:
More to come - I love your work! Thanks for making transitioning to Mastodon easier.
